imagetest/testworkflow_test.go

```diff
@@ -16,8 +16,222 @@ package imagetest
+		t.Error("StopInstances step is missing")
+	}
+	if len(step.StopInstances.Instances) != 1 {
+		t.Error("StopInstances step is malformed")
+	}
+	if step.StopInstances.Instances[0] != "vmname" {
+		t.Error("StopInstances step is malformed")
+	}
+	stepFromWF, ok := twf.wf.Steps["stop-stepname"]
+	if !ok || step != stepFromWF {
+		t.Error("step was not correctly added to workflow")
+	}
+}
+
+func TestAddWaitStep(t *testing.T) {
+	twf, err := NewTestWorkflow("name", "image")
+	if err != nil {
+		t.Errorf("failed to create test workflow: %v", err)
+	}
+	if twf.wf == nil {
+		t.Error("test workflow is malformed")
+	}
+	step, err := twf.addWaitStep("stepname", "vmname", false)
+	if err != nil {
+		t.Errorf("failed to add wait step to test workflow: %v", err)
+	}
+	if step.WaitForInstancesSignal == nil {
+		t.Error("WaitForInstancesSignal step is missing")
+	}
+	instancesSignal := []*daisy.InstanceSignal(*step.WaitForInstancesSignal)
+	if len(instancesSignal) != 1 {
+		t.Error("waitInstances step is malformed")
+	}
+	if instancesSignal[0].Name != "vmname" {
+		t.Error("waitInstances step is malformed")
+	}
+	if instancesSignal[0].SerialOutput.SuccessMatch != successMatch {
+		t.Error("waitInstances step is malformed")
+	}
+	if instancesSignal[0].Stopped {
+		t.Error("waitInstances step is malformed")
+	}
+	stepFromWF, ok := twf.wf.Steps["wait-stepname"]
+	if !ok || step != stepFromWF {
+		t.Error("step was not correctly added to workflow")
+	}
+}
+
+func TestAddWaitStoppedStep(t *testing.T) {
+	twf, err := NewTestWorkflow("name", "image")
+	if err != nil {
+		t.Errorf("failed to create test workflow: %v", err)
+	}
+	if twf.wf == nil {
+		t.Error("test workflow is malformed")
+	}
+	step, err := twf.addWaitStep("stepname", "vmname", true)
+	if err != nil {
+		t.Errorf("failed to add wait step to test workflow: %v", err)
+	}
+	if step.WaitForInstancesSignal == nil {
+		t.Error("WaitForInstancesSignal step is missing")
+	}
+	instancesSignal := []*daisy.InstanceSignal(*step.WaitForInstancesSignal)
+	if len(instancesSignal) != 1 {
+		t.Error("waitInstances step is malformed")
+	}
+	if instancesSignal[0].Name != "vmname" {
+		t.Error("waitInstances step is malformed")
+	}
+	if instancesSignal[0].SerialOutput != nil {
+		t.Error("waitInstances step is malformed")
+	}
+	if !instancesSignal[0].Stopped {
+		t.Error("waitInstances step is malformed")
+	}
+	stepFromWF, ok := twf.wf.Steps["wait-stepname"]
+	if !ok || step != stepFromWF {
+		t.Error("step was not correctly added to workflow")
+	}
+}
+
+func TestAppendCreateDisksStep(t *testing.T) {
+	twf, err := NewTestWorkflow("name", "image")
+	if err != nil {
+		t.Errorf("failed to create test workflow: %v", err)
+	}
+	if twf.wf == nil {
+		t.Error("test workflow is malformed")
+	}
+	step, err := twf.appendCreateDisksStep("diskname")
+	if err != nil {
+		t.Errorf("failed to add wait step to test workflow: %v", err)
+	}
+	if step.CreateDisks == nil {
+		t.Error("CreateDisks step is missing")
```

Comment:
Consider using `t.Fatal` here instead, as your test will otherwise panic on line 160. Not the end of the world if it panics (it's still considered a failure), but it makes the test logs harder to read.